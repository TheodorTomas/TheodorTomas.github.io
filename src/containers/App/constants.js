// TODO: move to styled-component theme.
export const COLORS = {
  darkGrey: {
    hex: '#4a4a4a',
    rgba: alpha => `rgba(74,74,74,${alpha})`,
  },
  blueGrey: {
    hex: '#95a3b3',
    rgba: alpha => `rgba(149,163,179,${alpha})`,
  },
  lightGrey: {
    hex: '#d3d3d3',
  },
  green: {
    hex: '#78c8b4',
  },
};
