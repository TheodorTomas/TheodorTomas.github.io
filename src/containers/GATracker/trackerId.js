// Ideally keys should be on a server due to this being public repository but this key
// does not need to be secure and is useful during development. Therefor it is stored here.
const getTrackerId = () => process.env.NODE_ENV === 'production' ? 'UA-109205585-1' : 'UA-000000000-0';
export default getTrackerId;
