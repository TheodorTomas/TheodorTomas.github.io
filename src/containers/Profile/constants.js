export const bio = ['JavaScript', 'HTML5', 'CSS3', 'React', 'Angular', 'Ionic', 'Node'];

export const icons = [
  { href: 'https://linkedin.com/in/TheodorTomas', title: 'www.linkedin.com/TheodorTomas', className: 'fa fa-linkedin', hoverColor: '#0077b5' },
  { href: 'https://github.com/TheodorTomas', title: 'www.github.com/TheodorTomas', className: 'fa fa-github', hoverColor: '#6cc644' },
  { href: 'https://twitter.com/TheodorTomas', title: 'www.twitter.com/TheodorTomas', className: 'fa fa-twitter', hoverColor: '#1da1f2' },
];

export const contactInfo = {
  href: 'mailto:theodortomas@gmail.com',
  title: 'Contact me',
  text: 'theodortomas@gmail.com',
};
