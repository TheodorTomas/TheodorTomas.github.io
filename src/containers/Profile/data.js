import linkedin from './media/linkedin';
import github from './media/github';
import twitter from './media/twitter';

export const icons = [
  {
    svg: linkedin,
    href: 'https://linkedin.com/in/TheodorTomas',
    title: 'www.linkedin.com/TheodorTomas',
  },
  {
    svg: github,
    href: 'https://github.com/TheodorTomas',
    title: 'www.github.com/TheodorTomas',
  },
  {
    svg: twitter,
    href: 'https://twitter.com/TheodorTomas',
    title: 'www.twitter.com/TheodorTomas',
  },
];

export const bio = ['JavaScript', 'HTML', 'CSS', 'React', 'Angular', 'Ionic', 'Node', 'Webpack'];

export const contactInfo = {
  href: 'mailto:theodortomas@gmail.com',
  title: 'Contact me',
  text: 'theodortomas@gmail.com',
};
