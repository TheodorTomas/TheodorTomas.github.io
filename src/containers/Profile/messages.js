/**
 * ProfilePage Messages
 *
 * This contains all the text for the ProfilePage container.
 */

const messages = {
  h1: 'Hello! My name is Theodór Tómas',
  h2: 'Software Engineer',
};

export default messages;
