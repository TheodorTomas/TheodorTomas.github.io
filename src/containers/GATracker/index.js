/**
 * Initializes Google analytics and sets the page view to '/'.
 */
import React from 'react';
import * as ReactGA from 'react-ga';
import { TRACKER_ID } from '../../constants/analytics';

export class GATracker extends React.Component {
  componentDidMount() {
    ReactGA.initialize(TRACKER_ID, { debug: false });
    ReactGA.set({ pathname: '/' });
    ReactGA.pageview('/');
  }

  render() {
    return null;
  }
}

export default GATracker;
