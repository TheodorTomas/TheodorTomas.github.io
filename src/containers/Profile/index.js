/**
 * Profile container, containing what data should be used with which components.
 */
import React from 'react';
import Styled from 'styled-components';
import Icons from 'components/Icons';
import Bio from 'components/Bio';
import Header from 'components/Header';
import EmailLink from 'components/EmailLink';
import Photo from 'components/Photo';
import { COLORS } from 'containers/App/constants';
import messages from './messages';
import photo from './images/photo.jpg';
import { bio, icons, contactInfo } from './constants';

const Wrapper = Styled.article`
  /* Flex-box */
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  
  /* Sizing */
  margin: auto;
  padding: 32px;
  width: 50%;
  min-width: calc(150px + (2 * 32px) + 10vw);
  
  /* Misc */
  position: relative;
  z-index: 1;
  background-color: ${COLORS.darkGrey.rgba(0.75)};
  border: 1px solid ${COLORS.darkGrey.rgba(0.75)};
  border-radius: 4px;
  box-shadow: 2px 2px 2px 1px rgba(0, 0, 0, 0.2);
  
  opacity: 0;
  
  /* Animations */
  @keyframes popIn {
    0% {
      transform: scale(0.95) translateZ(0);
      opacity: 0;
    }
    
    100% {
      transform: scale(1) translateZ(0);
      opacity: 1;
    }
  }
  
  animation: popIn 1s cubic-bezier(0.175, 0.885, 0.32, 1.275) forwards 250ms;
`;

const Profile = () => (
  <Wrapper>
    <Photo src={photo} alt="Profile Photo" />
    <Header h1={messages.h1} h2={messages.h2} />
    <Bio bio={bio} />
    <Icons icons={icons} />
    <EmailLink {...contactInfo} />
  </Wrapper>
);

export default Profile;
