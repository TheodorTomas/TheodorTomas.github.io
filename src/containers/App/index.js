/**
 * App
 *
 * Returns an array of components.
 */
import React from 'react';
import Profile from '../Profile';
import GATracker from '../GATracker';
import GlobalStyles from '../../global-styles';

export const App = () => ([
  <GlobalStyles key="global-styles" />,
  <GATracker key="ga-tracker" />,
  <Profile key="profile" />,
]);

export default App;
