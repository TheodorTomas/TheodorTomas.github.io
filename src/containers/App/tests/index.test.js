import React from 'react';
import Helmet from 'react-helmet';
import { shallow } from 'enzyme';
import GATracker from 'containers/GATracker';
import Profile from 'containers/Profile';
import App from '../index';

describe('<App />', () => {
  const renderComponent = (props = {}) => shallow(<App {...props} />);

  it('should contain a <Helmet> component', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.find(Helmet).length).toBe(1);
  });

  it('should contain a <GATracker> component', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.contains(<GATracker />)).toBe(true);
  });

  it('should contain a <Route> component to the profile component', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.contains(<Profile />)).toBe(true);
  });
});
