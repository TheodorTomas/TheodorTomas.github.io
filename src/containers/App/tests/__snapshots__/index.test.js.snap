// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render matching snapshot 1`] = `
.c5 .icon {
  fill: #95a3b3;
  cursor: pointer;
  margin: 0 1em;
  -webkit-transition: fill 0.25s ease-in-out,width 0.25s ease-in-out,height 0.25s ease-in-out;
  transition: fill 0.25s ease-in-out,width 0.25s ease-in-out,height 0.25s ease-in-out;
}

.c5 .linkedin:visited,
.c5 .linkedin:hover,
.c5 .linkedin:active {
  fill: #0077b5;
}

.c5 .github:visited,
.c5 .github:hover,
.c5 .github:active {
  fill: #6cc644;
}

.c5 .twitter:visited,
.c5 .twitter:hover,
.c5 .twitter:active {
  fill: #1da1f2;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0.75em 0 1em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #d3d3d3;
}

.c3 span {
  line-height: 100%;
  margin: 0.25em;
}

.c2 {
  margin: 1em 0 0.75em;
  text-align: center;
  color: #78c8b4;
}

.c2 h1,
.c2 h2 {
  margin: 0;
  line-height: 100%;
}

.c2 h2 {
  margin-top: 0.25em;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  line-height: 100%;
  text-align: center;
  color: #95a3b3;
  -webkit-transition: color 0.25s ease-in-out,font 0.25s ease-in-out;
  transition: color 0.25s ease-in-out,font 0.25s ease-in-out;
  cursor: pointer;
}

.c6:visited,
.c6:hover,
.c6:active {
  color: #78c8b4;
}

.c1 {
  width: 150px;
  height: 150px;
  margin: auto;
  background-color: #8f8c86;
  overflow: hidden;
  -webkit-transition: border-color 0.25s ease-in-out,-webkit-transform 0.25s ease-in-out;
  -webkit-transition: border-color 0.25s ease-in-out,transform 0.25s ease-in-out;
  transition: border-color 0.25s ease-in-out,transform 0.25s ease-in-out;
  border-radius: 50%;
  border: 2px solid;
  border-color: rgba(74,74,74,0.5);
  box-shadow: 2px 2px 2px 1px rgba(0,0,0,0.2);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: auto;
  padding: 32px;
  width: 50%;
  min-width: calc(150px + (2 * 32px) + 10vw);
  max-width: 500px;
  position: relative;
  z-index: 1;
  background-color: rgba(74,74,74,0.75);
  border: 1px solid rgba(74,74,74,0.75);
  border-radius: 4px;
  box-shadow: 2px 2px 2px 1px rgba(0,0,0,0.2);
  opacity: 0;
  -webkit-animation: popIn 1s cubic-bezier(0.175,0.885,0.32,1.275) forwards 250ms;
  animation: popIn 1s cubic-bezier(0.175,0.885,0.32,1.275) forwards 250ms;
}

@media only screen and (max-width:450px) {
  .c5 .icon {
    width: 2em;
    height: 2em;
  }
}

<article
  className="c0"
>
  <video
    autoPlay={true}
    className="c1"
    muted={true}
    onClick={[Function]}
    title=""
  >
    <source
      src="profile.mp4"
      type="video/mp4"
    />
  </video>
  <header
    className="c2"
  >
    <h1>
      <span>
        Hello! My name is Theodór Tómas
      </span>
    </h1>
    <h2>
      <span>
        Software Engineer
      </span>
    </h2>
  </header>
  <div
    className="c3"
  >
    <span>
      ~/
      JavaScript
    </span>
    <span>
      ~/
      HTML
    </span>
    <span>
      ~/
      CSS
    </span>
    <span>
      ~/
      React
    </span>
    <span>
      ~/
      Angular
    </span>
    <span>
      ~/
      Ionic
    </span>
    <span>
      ~/
      Node
    </span>
    <span>
      ~/
      Webpack
    </span>
  </div>
  <div
    className="c4"
  >
    <a
      className="c5"
      href="https://linkedin.com/in/TheodorTomas"
      onClick={[Function]}
      rel="noreferrer noopener"
      target="_blank"
      title="www.linkedin.com/TheodorTomas"
    >
      <svg
        className="icon linkedin"
        height="32"
        viewBox="0 0 1792 1792"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M477 625v991h-330v-991h330zm21-306q1 73-50.5 122t-135.5 49h-2q-82 0-132-49t-50-122q0-74 51.5-122.5t134.5-48.5 133 48.5 51 122.5zm1166 729v568h-329v-530q0-105-40.5-164.5t-126.5-59.5q-63 0-105.5 34.5t-63.5 85.5q-11 30-11 81v553h-329q2-399 2-647t-1-296l-1-48h329v144h-2q20-32 41-56t56.5-52 87-43.5 114.5-15.5q171 0 275 113.5t104 332.5z"
        />
      </svg>
    </a>
    <a
      className="c5"
      href="https://github.com/TheodorTomas"
      onClick={[Function]}
      rel="noreferrer noopener"
      target="_blank"
      title="www.github.com/TheodorTomas"
    >
      <svg
        className="icon github"
        height="32"
        viewBox="0 0 1792 1792"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M896 128q209 0 385.5 103t279.5 279.5 103 385.5q0 251-146.5 451.5t-378.5 277.5q-27 5-40-7t-13-30q0-3 .5-76.5t.5-134.5q0-97-52-142 57-6 102.5-18t94-39 81-66.5 53-105 20.5-150.5q0-119-79-206 37-91-8-204-28-9-81 11t-92 44l-38 24q-93-26-192-26t-192 26q-16-11-42.5-27t-83.5-38.5-85-13.5q-45 113-8 204-79 87-79 206 0 85 20.5 150t52.5 105 80.5 67 94 39 102.5 18q-39 36-49 103-21 10-45 15t-57 5-65.5-21.5-55.5-62.5q-19-32-48.5-52t-49.5-24l-20-3q-21 0-29 4.5t-5 11.5 9 14 13 12l7 5q22 10 43.5 38t31.5 51l10 23q13 38 44 61.5t67 30 69.5 7 55.5-3.5l23-4q0 38 .5 88.5t.5 54.5q0 18-13 30t-40 7q-232-77-378.5-277.5t-146.5-451.5q0-209 103-385.5t279.5-279.5 385.5-103zm-477 1103q3-7-7-12-10-3-13 2-3 7 7 12 9 6 13-2zm31 34q7-5-2-16-10-9-16-3-7 5 2 16 10 10 16 3zm30 45q9-7 0-19-8-13-17-6-9 5 0 18t17 7zm42 42q8-8-4-19-12-12-20-3-9 8 4 19 12 12 20 3zm57 25q3-11-13-16-15-4-19 7t13 15q15 6 19-6zm63 5q0-13-17-11-16 0-16 11 0 13 17 11 16 0 16-11zm58-10q-2-11-18-9-16 3-14 15t18 8 14-14z"
        />
      </svg>
    </a>
    <a
      className="c5"
      href="https://twitter.com/TheodorTomas"
      onClick={[Function]}
      rel="noreferrer noopener"
      target="_blank"
      title="www.twitter.com/TheodorTomas"
    >
      <svg
        className="icon twitter"
        height="32"
        viewBox="0 0 1792 1792"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1684 408q-67 98-162 167 1 14 1 42 0 130-38 259.5t-115.5 248.5-184.5 210.5-258 146-323 54.5q-271 0-496-145 35 4 78 4 225 0 401-138-105-2-188-64.5t-114-159.5q33 5 61 5 43 0 85-11-112-23-185.5-111.5t-73.5-205.5v-4q68 38 146 41-66-44-105-115t-39-154q0-88 44-163 121 149 294.5 238.5t371.5 99.5q-8-38-8-74 0-134 94.5-228.5t228.5-94.5q140 0 236 102 109-21 205-78-37 115-142 178 93-10 186-50z"
        />
      </svg>
    </a>
  </div>
  <a
    className="c6"
    href="mailto:theodortomas@gmail.com"
    onClick={[Function]}
    target="_self"
    title="Contact me"
  >
    theodortomas@gmail.com
  </a>
</article>
`;
